docs: integrate webserver configuration (#36553)
angular/angular: aio/content/guide/i18n.md

You can also use the CLI `--baseHref` option with [`ng build`](cli/build "CLI reference for ng build") to declare the base `href` at compile time.

### Configuring servers

> Typical deployment of multiple languages will serve each language from a different subdirectory.
> The user will be redirected to the preferred language defined in the browser (via the
> `Accept-Language` HTTP header), or it will fallback to a default language if a preferred language
> is not found. And the user will still be able to change language by navigating to other subdirectories,
> most likely via a some kind of menu implemented in the application.

Reviewer: ```suggestion
Typical deployment of multiple languages serve each language from a different subdirectory.
Users are redirected to the preferred language defined in the browser using the
`Accept-Language` HTTP header. If the user has not defined a preferred language, the browser uses the default language. Users can change the language by navigating to other subdirectories, which often occurs using a menu implemented in the application.
```

In general, avoid using "will" in sentences. Instead, use present tense.

Also, words such as "it" or "this" on their own are ambiguous. For example, "...it will fallback. " What does "it" refer to? I made a guess here--let me know if it's not accurate.

Otherwise, I made a few changes to make the text more concise.

> For other general advice on how to deploy apps to a remote server, see [Deployment](guide/deployment "Deployment guide").

Reviewer: ```suggestion
For more information on how to deploy apps to a remote server, see [Deployment](guide/deployment "Deployment guide").
```

Updated to use standard phrasing.

#### Nginx

> The Nginx configuration could look something like the following:

Reviewer: ```suggestion
The following is an example of an Nginx configuration.
```

```
http {

    # Browser preferred language detection (does NOT require AcceptLanguageModule)
    map $http_accept_language $accept_language {
        ~*^de de;
        ~*^fr fr;
        ~*^en en;
    }
    # ...
}

server {
    listen 80;
    server_name localhost;
    root /www/data;

    # Fallback to default language if no preference defined by browser 
    if ($accept_language ~ "^$") {
        set $accept_language "fr";
    }
    
    # Redirect "/" to Angular app in browser's preferred language
    rewrite ^/$ /$accept_language permanent;

    # Everything under the Angular app is always redirected to Angular in the correct language
    location ~ ^/(fr|de|en) {
        try_files $uri /$1/index.html?$args;
    }
    # ...
}
```

#### Apache

> The Apache configuration could look something like the following:

Reviewer: ```suggestion
The following is an example of an Apache configuration.
```

### Deployment for multiple locales

> Typical deployment of multiple languages will serve each language from a different subdirectory. The user will be redirected to the preferred language defined in the browser, or `fr` if a preferred language is not found. And the user will still be able to change language by navigating to other subdirectories.

Reviewer 2: > or `fr` if a preferred language is not found

This implies that `fr` is always the default. Can you re-word this to indicate that this is not always the case?

Also I note that this is one very long line! Can you add some line-breaks to keep the lines down to around 100 chars in length?

Author: resolved in rebase

 
#### Nginx

The Nginx configuration could look something like the following:

```
http {

>     # Browser preferred language detection (does NOT requires AcceptLanguageModule)

Reviewer 2: ```suggestion
    # Browser preferred language detection (does NOT require AcceptLanguageModule)
```
